### DISCUSSION TITLE
prepHumanModelForftINIT issue when working with MouseGEM

### DISCUSSION THREAD

**Discussion by rsango6** [SysBioChalmers/Human-GEM#691] [Q&A] [answered]
Hi there,

I am trying to generate ftINIT models with MouseGEM and am following a tutorial on https://sysbiochalmers.github.io/Human-GEM-guide/gem_extraction/

In the preparation step, I am trying to generate a prepData struct with the following code:

```
prepData = prepHumanModelForftINIT(mouseGEM, false, ...
    '/path/to/Human-GEM/data/metabolicTasks/metabolicTasks_Essential.txt', ...
    '/path/to/Mouse-GEM/model/reactions.tsv');

```

The error I get is the following:

```
Step 3: Check tasks (~10 min)
Attempt to grow array along ambiguous dimension.

Error in closeModel (line 50)
        closedModel.b(numel(closedModel.b)+1)=0;

Error in prepINITModel (line 82)
    bModel = closeModel(cModel);

Error in prepHumanModelForftINIT (line 133)
prepDataHumanGEM = prepINITModel(m, taskStruct, spontRxnNames, convertGenes, customRxnsToIgnore, 'e');

```

I have tried the same thing with HumanGEM, and it successfully passes the third step when it checks metabolic tasks. Any idea why MouseGEM can't do the same? I suspect it is something to do with `MetabolicTasks.txt` file but not sure exactly what. 

Many thanks!

**Comment by rsango6** [comment, marked as answer]
I was able to solve this:

I had to transform the model into a closed-form, i.e. `model = addBoundaryMets(model)`. I can now proceed through the third step and now can go to other steps in model generation.

**Reply by haowang-bioinfo** [reply]
very good
